PKHeX.WinForms/Program.cs

```diff
@@ -29,16 +29,33 @@ private static void Main()
 
             try
             {
-                if (GetFrameworkVersion() >= 393295)
+                if (IsOnWindows())
                 {
-                    StartPKHeX();
+                    if (GetFrameworkVersion() >= 393295)
+                    {
+                        StartPKHeX();
+                    }
+                    else
+                    {
+                        // Todo: make this translatable
+                        MessageBox.Show(".NET Framework 4.6 needs to be installed for this version of PKHeX to run.", "PKHeX Error", MessageBoxButtons.OK, MessageBoxIcon.Stop);
+                        Process.Start(@"https://www.microsoft.com/download/details.aspx?id=48130");
+                    }
                 }
                 else
                 {
-                    // Todo: make this translatable
-                    MessageBox.Show(".NET Framework 4.6 needs to be installed for this version of PKHeX to run.", "PKHeX Error", MessageBoxButtons.OK, MessageBoxIcon.Stop);
-                    Process.Start(@"https://www.microsoft.com/download/details.aspx?id=48130");
+                    //CLR Version 4.0.30319.42000 is equivalent to .NET Framework version 4.6
+                    if ((Environment.Version.CompareTo(Version.Parse("4.0.30319.42000"))) >= 0)
```

Comment:
I'm still testing in VMs, but this works in Windows 10, even after locally switching the project to .Net 4.6.1 and .Net 4.6.2.  You might not even need the OS check.